### ISSUE TITLE
Is hmatrix thread safe?

### ISSUE THREAD

**Issue by XinhuaZhang** [albertoruiz/hmatrix#212]
Hi,

I wonder if I could run parMap with +RTS -N option to use hmatrix correctly now? I remember that when I tried this a long time ago, it gave me weird results. Since hmatrix uses external library, I guess that it was not thread safe.

**Comment by albertoruiz** [comment]
Yes, hmatrix-base is thread safe (assuming no bugs). It only uses blas-lapack and custom C code without state. We had some problems in the past with random numbers, https://github.com/albertoruiz/hmatrix/issues/97, but I hope this is now solved.
If you find something strange please make a bug report.
